linkerd/proxy/spire-client/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+mod api;
+
+pub use api::{Api, SvidUpdate};
+use linkerd_error::{Error, Result};
+use linkerd_identity::Credentials;
+use linkerd_identity::Id;
+use std::fmt::Display;
+use std::time::{Duration, UNIX_EPOCH};
+use tokio::sync::watch;
+use tower::Service;
+use tracing::error;
+
+pub struct Spire {
+    id: Id,
+}
+
+// === impl Spire ===
+
+impl Spire {
+    pub fn new(id: Id) -> Self {
+        Self { id }
+    }
+
+    pub async fn run<C, S>(self, credentials: C, mut client: S)
+    where
+        C: Credentials,
+        S: Service<(), Response = tonic::Response<watch::Receiver<SvidUpdate>>>,
+        S::Error: Into<Error> + Display,
+    {
+        match client.call(()).await {
```

Comment:
I'd like to talk a little bit more about the error handling in this task, but to satisfy the Service contract, we need to await readiness with something like `client.ready().await?.call(().await` (using the ServiceExt trait to get the ready() call).